src/lib/actions.js

```diff
@@ -195,10 +195,17 @@ function getCanonicalPath(path) {
 
 export const checkUserPreferredLanguage = store.action(
   ({userPreferredLanguage}) => {
-    if (!userPreferredLanguage) {
-      return {
-        userPreferredLanguage: cookies.get('preferred_lang'),
-      };
+    userPreferredLanguage =
+      // Use currently set language.
+      userPreferredLanguage ||
+      // Or check in the url.
+      lang.getLanguageFromPath(location.pathname) ||
```

Comment:
If I switch to PL using the dropdown but then type "web.dev/en/blah", I get the English page then further links switch back to Polish.

Reply:
This happens currently because in the repo we have now an incomplete pl directory. English page is a fallback for Polish urls that might be missing in /pl/ directory. If you go to a page that has currently a polish version - e.g. /accessible/ and then switch language, you should see the correct behavior.

Reply:
I pushed the full language content to staging: https://web-dev-staging.uc.r.appspot.com/ so you can safely test there.